File: csharp-xunit/WordCount.Tests/Session2c_WordCounterTest.cs
using NHamcrest;
using NHamcrest.XUnit;

namespace Org.Codecop.WordCount.Tests
{
    /// <summary>Bonus Session 2c: WordCounterTest - Assertions with Hamcrest.</summary>
    /// <seealso>"https://github.com/nhamcrest/NHamcrest"</seealso>
    public class Session2c_WordCounterTest
    {

        // TODO Add the proper assertions to complete the tests using Hamcrest.
        // The test name explains what needs to be asserted.

        [Xunit.Fact]
        public void ShouldCountNumberOfWords()
        {
            var counter = new WordCounter("Keep the bar green to keep the code clean.");
            Xunit.Assert.Equal(9, counter.NumberOfWords());
            // Hamcrest improves readability:
            // TODO Check that counter.NumberOfWords(), EqualTo 9.
        }

        [Xunit.Fact]
        public void ShouldContainUniqueWord()
        {
            var counter = new WordCounter("green bar green hat");
            Xunit.Assert.Contains(counter.UniqueWords(), w => w == "bar");
            // Hamcrest improves readability:
            // TODO Check that counter.UniqueWords(), Has.Item EqualTo "bar".
        }

        [Xunit.Fact]
        public void ShouldNotContainUniqueWord()
        {
            var counter = new WordCounter("green bar green hat");
            Xunit.Assert.DoesNotContain(counter.UniqueWords(), w => w == "foo");
            // Hamcrest improves readability:
            // TODO Check that counter.UniqueWords(), Not Has.Item EqualTo "foo".
        }

        [Xunit.Fact]
        public void ShouldFindNumberOfUniqueWords()
        {
            var counter = new WordCounter("green bar green hat");
            Xunit.Assert.Equal(3, counter.UniqueWords().Length);
            // Hamcrest improves readability:
            // TODO Check that counter.UniqueWords(), OfLength 3.
        }
    }
}

File: php-phpunit/test/Session2c_WordCounterTest.php
<?php

use \Wordcount\WordCounter;

/**
 * Bonus Session 2c: WordCounterTest - Assertions with Hamcrest.
 * See https://github.com/hamcrest/hamcrest-php
 */
class Session2c_WordCounterTest extends \PHPUnit_Framework_TestCase {

    // TODO Add the proper assertions to complete the tests using Hamcrest.

    /** @test */
    function shouldCountNumberOfWords() {
        $counter = new WordCounter("Keep the bar green to keep the code clean.");
        $this->assertEquals(9, $counter->numberOfWords()); 
        // Hamcrest improves readability:
        // TODO Check that $counter->numberOfWords(), equalTo 9.
    }

    /** @test */
    function shouldContainUniqueWord() {
        $counter = new WordCounter("green bar green hat");
        $this->assertContains("bar", $counter->uniqueWords()); 
        // Hamcrest improves readability:
        // TODO Check that $counter->uniqueWords(), hasItem "bar".
    }

    /** @test */
    function shouldNotContainUniqueWord() {
        $counter = new WordCounter("green bar green hat");
        $this->assertNotContains("foo", $counter->uniqueWords()); 
        // Hamcrest improves readability:
        // TODO Check that $counter->uniqueWords(), not hasItem "foo".
    }

    /** @test */
    function shouldFindNumberOfUniqueWords() {
        $counter = new WordCounter("green bar green hat");
        $this->assertCount(3, $counter->uniqueWords()); 
        // Hamcrest improves readability:
        // TODO Check that $counter->uniqueWords(), arrayWithSize 3.
    }

}
